pkg/needacert/needacert.go

```diff
@@ -309,15 +345,20 @@ func (h *handler) generateSecret(service *corev1.Service) (*corev1.Secret, error
 }
 
 func (h *handler) updateSecret(owner runtime.Object, secret *corev1.Secret, dnsNames []string) (*corev1.Secret, error) {
-	tlsCert, err := tls.X509KeyPair(secret.Data[corev1.TLSCertKey], secret.Data[corev1.TLSPrivateKeyKey])
-	if err != nil || len(tlsCert.Certificate) == 0 {
+	cert, err := parseCert(secret)
```

Comment:
Non-blocking nitpick: is this call here necessary?

I see `parseCert` being called here and then again as the first thing inside of `scheduleNextCertCheck` few lines below.

I wonder if it would read better to only do the parsing inside of `scheduleNextCertCheck`.